runtime/auth/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/awssigning/internal/AwsChunkedUtil.kt

```diff
@@ -66,7 +66,10 @@ public val AwsSigningConfig.useAwsChunkedEncoding: Boolean
 public fun HttpRequestBuilder.setAwsChunkedHeaders() {
     headers.append("Content-Encoding", "aws-chunked")
     headers["Transfer-Encoding"] = "chunked"
-    headers["X-Amz-Decoded-Content-Length"] = body.contentLength!!.toString()
+    headers["X-Amz-Decoded-Content-Length"] = checkNotNull(headers["Content-Length"] ?: body.contentLength?.toString()) {
+        "Expected \"Content-Length\" header or `body.contentLength` to be set for aws-chunked"
+    }
```

Comment:
**Nit**: `checkNotNull` throws `IllegalStateException` in the null case. We should probably throw `ClientException` or some subclass of that.